app/src/main/java/org/vss/VSSApplication.java

```diff
@@ -1,10 +1,28 @@
 package org.vss;
 
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import jakarta.inject.Inject;
 import jakarta.ws.rs.ApplicationPath;
+import org.glassfish.hk2.api.ServiceLocator;
 import org.glassfish.jersey.server.ResourceConfig;
+import org.jvnet.hk2.guice.bridge.api.GuiceBridge;
+import org.jvnet.hk2.guice.bridge.api.GuiceIntoHK2Bridge;
+import org.vss.guice.BaseModule;
 
 @ApplicationPath("/")
 public class VSSApplication extends ResourceConfig {
-  public VSSApplication() {
+
+  @Inject
+  public VSSApplication(ServiceLocator serviceLocator) {
+    packages("org.vss");
+    Injector injector = Guice.createInjector(new BaseModule());
+    initGuiceIntoHK2Bridge(serviceLocator, injector);
+  }
+
+  private void initGuiceIntoHK2Bridge(ServiceLocator serviceLocator, Injector injector) {
+    GuiceBridge.getGuiceBridge().initializeGuiceBridge(serviceLocator);
+    GuiceIntoHK2Bridge guiceBridge = serviceLocator.getService(GuiceIntoHK2Bridge.class);
+    guiceBridge.bridgeGuiceInjector(injector);
   }
```

Comment:
Having no idea how this works, could you give a TL; DR context in the commit message?